pandas/core/internals/blocks.py

```diff
@@ -720,7 +720,10 @@ def copy(self, deep=True):
         """ copy constructor """
         values = self.values
         if deep:
-            values = values.copy()
+            if self.is_extension:
+                values = values.copy(deep=True)
+            else:
+                values = values.copy()
```

Comment:
Why is a deep-copy needed here?

Reply:
@gfyoung just pushed addressing most of your comments.  for this one: because without deep=True, it isn't a "real" copy.  i.e. `ExtensionArray.copy(deep=True)` behaves the same as `np.ndarray.copy()`.